contracts/modules/licensing/BasePolicyFrameworkManager.sol

```diff
@@ -4,26 +4,82 @@ pragma solidity 0.8.23;
 // external
 import { ERC165 } from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
 import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 // contracts
 import { IPolicyFrameworkManager } from "../../interfaces/modules/licensing/IPolicyFrameworkManager.sol";
-import { LicensingModuleAware } from "../../modules/licensing/LicensingModuleAware.sol";
+import { ILicensingModule } from "../../interfaces/modules/licensing/ILicensingModule.sol";
+import { Errors } from "../../lib/Errors.sol";
 
 /// @title BasePolicyFrameworkManager
+/// TODO: If we want to open this, we need an upgradeable and non-upgradeable Base version, or just promote
+/// the IPolicyFrameworkManager in the docs.
 /// @notice Base contract for policy framework managers.
-abstract contract BasePolicyFrameworkManager is IPolicyFrameworkManager, ERC165, LicensingModuleAware {
-    /// @notice Returns the name to be show in license NFT (LNFT) metadata
-    string public override name;
+abstract contract BasePolicyFrameworkManager is IPolicyFrameworkManager, ERC165, Initializable {
+    /// @dev Storage for BasePolicyFrameworkManager
+    /// @param name The name of the policy framework manager
+    /// @param licenseTextUrl The URL to the off chain legal agreement template text
+    /// @custom:storage-location erc7201:story-protocol.BasePolicyFrameworkManager
+    struct BasePolicyFrameworkManagerStorage {
+        string name;
+        string licenseTextUrl;
+    }
 
-    /// @notice Returns the URL to the off chain legal agreement template text
-    string public override licenseTextUrl;
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.BasePolicyFrameworkManager")) - 1))
+    // & ~bytes32(uint256(0xff));
+    bytes32 private constant BasePolicyFrameworkManagerStorageLocation =
+        0xa55803740ac9329334ad7b6cde0ec056cc3ba32125b59c579552512bed001f00;
```

Comment:
Is there a tool or command line to easily calculate or verify the storage location?

Reply:
This works too:
```bash
name="story-protocol.MockLicenseRegistryV2"

k1=$(cast keccak $name)
read d1 <<< "$(bc -l <<< "$(cast to-dec $k1) - 1")"
a1=$(cast abi-encode "f(uint256)" $d1)
k2=$(cast keccak $a1)

python -c "print(hex($k2 & 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00))"
```

Reply:
Thanks @jdubpark